kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -237,14 +283,30 @@ translatePredicate translateUninterpreted predicate =
         Attribute.Sort { hook = Hook { getHook } } =
             sortAttributes tools sort
 
+data SmtEncoding variable = SmtEncoding
+    { smtName :: Text
+    , smtType :: SExpr
+    , boundVars :: [ElementVariable variable]
+    }
+    deriving (Eq, GHC.Generic, Show)
+
 -- ----------------------------------------------------------------
 -- Translator
-type VarContext p = Map p (SExpr, SExpr)
+data VarContext variable
```

Comment:
I think we should call this `TranslatorState`, or something like that, because it emphasizes that it's related to the translator, and because it doesn't only concern variables.